spec/unit/radius_ip_whitelist_spec.rb

```diff
@@ -0,0 +1,88 @@
+describe RadiusIpWhitelist do
+  let(:organisation) { create(:organisation) }
+  let(:location1) { create(:location, organisation: organisation) }
+  let(:location2) { create(:location, organisation: organisation) }
+
+  before do
+    create(:ip, address: '1.1.1.1', location: location1)
+    create(:ip, address: '1.2.2.1', location: location1)
+    create(:ip, address: '2.2.2.2', location: location2)
+
+    location1.update(radius_secret_key: 'radkey1')
+    location2.update(radius_secret_key: 'radkey2')
+  end
+
+  let(:configuration_result) { subject.execute }
+
+  describe '#execute' do
+    it 'returns the correct configuration' do
+      expect(configuration_result).to eq('client 1-1-1-1 {
+  ipaddr = 1.1.1.1
+  secret = radkey1
+}
+client 1-2-2-1 {
+  ipaddr = 1.2.2.1
+  secret = radkey1
+}
+client 2-2-2-2 {
+  ipaddr = 2.2.2.2
+  secret = radkey2
+}
+')
+    end
+  end
+
+  context 'with duplicates in the database' do
+    before do
+      create(:ip, address: '2.2.2.2', location: create(:location, organisation: create(:organisation)))
```

Comment:
I had expected this to be silently failing to create because of our "unique IP" restriction, and that the test was producing a false positive based on that.

Turns out... we don't actually validate on unique IPs in this app? That seems like a mistake on our part!

Reply:
Yes needs fixing at application database level.
Given this is of crucial importance, I'm happy with a guard here too

Reply:
I am very happy to have the test here as well!